zipline/utils/numpy_utils.py

```diff
@@ -1,7 +1,40 @@
 """
 Utilities for working with numpy arrays.
 """
+from numpy import (
+    broadcast,
+    busday_count,
+    datetime64,
+    dtype,
+    empty,
+    nan,
+    where
+)
 from numpy.lib.stride_tricks import as_strided
+from toolz import flip
+
+uint8_dtype = dtype('uint8')
+bool_dtype = dtype('bool')
+int64_dtype = dtype('int64')
+float64_dtype = dtype('float64')
+datetime64D_dtype = dtype('datetime64[D]')
+datetime64ns_dtype = dtype('datetime64[ns]')
+
+make_datetime64ns = flip(datetime64, 'ns')
+make_datetime64D = flip(datetime64, 'D')
+np_NaT = make_datetime64ns('NaT')
+
+_FILLVALUE_DEFAULTS = {
+    float64_dtype: nan,
+    datetime64ns_dtype: np_NaT,
+}
+
+
+def default_fillvalue_for_dtype(dtype):
+    """
+    Get the default fill value for `dtype`.
+    """
+    return _FILLVALUE_DEFAULTS[dtype]
```

Comment:
I can imagine the implementation becoming more complex in the future.  The fact that it's just a simple dict lookup right now is an implementation detail.